tests/storage_test.go

```diff
@@ -245,6 +245,105 @@ var _ = Describe("Storage", func() {
+				tests.DeletePvAndPvc(pvName)
+
+				By("Deleting NFS pod")
+				Expect(virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Delete(tests.NFSTargetName, &metav1.DeleteOptions{})).To(Succeed())
+				By("Waiting for NFS pod to disappear")
+				tests.WaitForPodToDisappearWithTimeout(tests.NFSTargetName, 120)
+			})
+			It("should start a VMI with virtiofs", func() {
+				hugepageSize := "2Mi"
+				memory := "64Mi"
+				guestMemory := "None"
+				// Start the VirtualMachineInstance with the PVC attached
+				By("Creating the  VMI")
+				vmi = tests.NewRandomVMIWithPVCFS(pvName)
+				vmi.Labels = map[string]string{
+					"debugLogs":          "true",
+					"virtiofsdDebugLogs": "true",
+				}
+				hugepageType := k8sv1.ResourceName(k8sv1.ResourceHugePagesPrefix + hugepageSize)
+
+				nodeWithHugepages := tests.GetNodeWithHugepages(virtClient, hugepageType)
+				if nodeWithHugepages == nil {
+					Skip(fmt.Sprintf("No node with hugepages %s capacity", hugepageType))
+				}
+				vmi.Spec.Affinity = &k8sv1.Affinity{
+					NodeAffinity: &k8sv1.NodeAffinity{
+						RequiredDuringSchedulingIgnoredDuringExecution: &k8sv1.NodeSelector{
+							NodeSelectorTerms: []k8sv1.NodeSelectorTerm{
+								{
+									MatchExpressions: []k8sv1.NodeSelectorRequirement{
+										{Key: "kubernetes.io/hostname", Operator: k8sv1.NodeSelectorOpIn, Values: []string{nodeWithHugepages.Name}},
+									},
+								},
+							},
+						},
+					},
+				}
+				vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse(memory)
+
+				vmi.Spec.Domain.Memory = &v1.Memory{
+					Hugepages: &v1.Hugepages{PageSize: hugepageSize},
+				}
+				if guestMemory != "None" {
+					guestMemReq := resource.MustParse(guestMemory)
+					vmi.Spec.Domain.Memory.Guest = &guestMemReq
+				}
+				vmi.Spec.Domain.Devices.Rng = &v1.Rng{}
+
+				// add userdata for guest agent and mount virtio-fs
+				fs := vmi.Spec.Domain.Devices.Filesystems[0]
+				virtiofsMountPath := fmt.Sprintf("/mnt/virtiof_%s", fs.Name)
+				mountVirtiofsCommands := fmt.Sprintf(`
+                                       mkdir %s
+                                       mount -t virtiofs %s %s
+                               `, virtiofsMountPath, fs.Name, virtiofsMountPath)
+				userData := fmt.Sprintf("%s\n%s", tests.GetGuestAgentUserData(), mountVirtiofsCommands)
+				tests.AddUserData(vmi, "cloud-init", userData)
+
+				vmi = tests.RunVMIAndExpectLaunchIgnoreWarnings(vmi, 180)
+
+				// Wait for cloud init to finish and start the agent inside the vmi.
+				tests.WaitAgentConnected(virtClient, vmi)
+
+				By("Checking that the VirtualMachineInstance console has expected output")
+				expecter, err := tests.LoggedInFedoraExpecter(vmi)
+				Expect(err).ToNot(HaveOccurred(), "Should be able to login to the Fedora VM")
+				defer expecter.Close()
+
+				By("Checking that virtio-fs is mounted")
+				_, err = expecter.ExpectBatch([]expect.Batcher{
+					&expect.BSnd{S: "ls -l %s/*disk* | wc -l\n"},
```

Comment:
What does `%s` refer to here? Nit: maybe use `ls -1`?

Reply:
@jean-edouard any chance you could see what did I do wrong here? This fails with the new expecter.